### PR TITLE
Initial post metadata handling (CWs, images)

### PR DESCRIPTION
**Pull request by snowkat**
Gonna try to make more of these larger features actual PRs.

This introduces content warning support in the post feed, and a gallery box to view image attachments.

Current status:
- [X] Warn user of content warnings
- [X] Allow user to show/hide post with CWs
- [X] Image gallery box in posts, split across rows
- [x] Clicking an image preview shows a lightbox with the full image
- [x] Figure out if I can use `` to show the original image size if the small no longer suffices

### CODE REVIEW

**Comment by vivlim** [comment]
looking at the build and deploy job, looks like it deployed to https://gray-water-0add53f1e-3.westus2.5.azurestaticapps.net/ but posting a comment with that link in the pr failed automatically

**Comment by vivlim** [comment]
It's a bit rough with images that are taller than they are wide, e.g. https://gray-water-0add53f1e-3.westus2.5.azurestaticapps.net/post/01J9A5WB94EF8VDF1FQED5VPS9 can take several screenfuls to scroll through if your window is wide enough, but that's not blocking & can be also fixed by constraining the maximum width of the Post Zone™️. overall looks great to me, nice work

**Comment by vivlim** [comment]
btw if you wanted to merge to main like this, you've got my 👍

**Review by vivlim — APPROVED** [review]

**Comment by snowkat** [comment]
I wanted to get the image lightboxes working first, which is now done! If it works fine for you, I'll merge it.

File: src/components/post/image-lightbox.tsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+};
+
+export interface ImageLightboxProps {
+    images: Array<Attachment>;
+    imageIndex?: number;
+    open?: boolean;
+    onOpenChange?: (isOpen: boolean) => void;
+}
+
+// HACK: there *has* to be a better way to do this, but this works for now
```
Comment (vivlim) [inline src/components/post/image-lightbox.tsx:35]: maaaaybe props? i haven't tried these but seems neat https://docs.solidjs.com/concepts/control-flow/dynamic#props

**Review by vivlim — APPROVED** [review]

**Comment by vivlim** [comment]
i attempted to resolve a conflict in the browser and if it still works i'll merge

**Pull request merged**